### PR TITLE
Contest System Improvements

### PR DESCRIPTION
**Pull request by nekodex**
Some refactoring work and adds the ability to show (and link to) entrants on contest results page.

### CODE REVIEW

**Comment by nekodex** [comment]
Travis will fail until #830 gets merged

File: app/Models/Contest.php [hunk 1]
```diff
@@ -89,9 +92,22 @@ public function state()
...
 
-    public function entryOrientation()
+    public function getEntryShapeAttribute()
     {
-        return $this->id === 5 ? 'landscape' : null;
+        if ($this->type !== 'art') {
+            return;
+        }
+
+        return $this->extra_options ? $this->extra_options['shape'] : 'square';
```
Comment (nanaya) [inline app/Models/Contest.php]: probably `$this->extra_options['shape'] ?? 'square'`

File: resources/assets/less/bem/contest-art-entry.less [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+      font-size: 20px;
+      white-space: nowrap;
+    }
+  }
+
+  &__result-votes {
+    font-size: 10px;
+    color: #8dacb6;
+    &--percentage {
+      font-size: 10px;
```
Comment (nanaya) [inline resources/assets/less/bem/contest-art-entry.less]: das coo

File: app/Models/Contest.php [hunk 3]
```diff
@@ -89,9 +92,22 @@ public function state()
         return 'over';
     }
 
-    public function entryOrientation()
+    public function getEntryShapeAttribute()
```
Comment (nanaya) [inline app/Models/Contest.php]: more like preview shape

**Pull request merged**